core/client/fs/src/main/java/alluxio/client/file/cache/PageStore.java

```diff
@@ -69,7 +85,64 @@ static PageStore create(AlluxioConfiguration conf) {
         throw new IllegalArgumentException(String.format("Unrecognized store type %s",
             storeType.name()));
     }
-    options.setRootDir(conf.get(PropertyKey.USER_CLIENT_CACHE_DIR));
+    String rootPath = conf.get(PropertyKey.USER_CLIENT_CACHE_DIR);
+    options.setRootDir(rootPath);
+    Path confPath = Paths.get(rootPath, CONF_FILE);
+    boolean canLoad = false;
+    if (Files.exists(confPath)) {
+      Properties properties = ConfigurationUtils.loadPropertiesFromFile(confPath.toString());
+      if (properties != null) {
+        AlluxioProperties alluxioProperties = new AlluxioProperties();
+        alluxioProperties.merge(properties, Source.DEFAULT);
+        AlluxioConfiguration cacheConf = new InstancedConfiguration(alluxioProperties);
+        // check store type
+        if (cacheConf.get(PropertyKey.USER_CLIENT_CACHE_STORE_TYPE).equals(
+            conf.get(PropertyKey.USER_CLIENT_CACHE_STORE_TYPE))
+            // check page size
+            && cacheConf.getBytes(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE)
+            == conf.getBytes(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE)
+            // check enough cache size
+            && cacheConf.getBytes(PropertyKey.USER_CLIENT_CACHE_SIZE)
+            <= conf.getBytes(PropertyKey.USER_CLIENT_CACHE_SIZE)
+            // check alluxio version
+            && cacheConf.get(PropertyKey.VERSION).equals(
+            conf.get(PropertyKey.VERSION))) {
+          LOG.info("Found recoverable local cache at {}", rootPath);
+          canLoad = true;
+        } else {
+          LOG.info("Found local cache at {} with incompatible configuration.", rootPath);
```

Comment:
Would be nice to know why its incompatible.

Reply:
Added individual checks and log messages.